File: lib/question.dart
class Question {
  final String questionText;
  final List<String> choices;
  final List<String> choicesHints;
  final int correctAnswerIndex;

  Question({
    required this.questionText,
    required this.choices,
    required this.choicesHints,
    required this.correctAnswerIndex,
  });
}

class ChoiceRecord {
  int key;
  String value;

  ChoiceRecord(this.key, this.value);
}

File: python_scripts/question.py
from dataclasses import dataclass
from typing import List


@dataclass
class Question:
    questionText: str
    choices: List[str]
    choicesHints: List[str]
    correctAnswerIndex: int = 0

    @staticmethod
    def to_quoted(items: List[str]):
        return ', '.join([f'"{p}"' for p in items])

    def to_dart(self):
        return f"""Question(
            questionText: "{self.questionText}",
            choices: [{self.to_quoted(self.choices)}],
            choicesHints: [{self.to_quoted(self.choicesHints)}],
            correctAnswerIndex: {self.correctAnswerIndex},
        )"""
